hrms/payroll/doctype/bulk_salary_structure_assignment/bulk_salary_structure_assignment.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) 2024, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from pypika import functions as fn
+
+import frappe
+from frappe import _
+from frappe.model.document import Document
+
+from hrms.hr.utils import notify_status
+from hrms.payroll.doctype.salary_structure.salary_structure import (
+	create_salary_structure_assignment,
+)
+
+
+class BulkSalaryStructureAssignment(Document):
+	@frappe.whitelist()
+	def get_employees(self, advanced_filters: list) -> list:
+		Assignment = frappe.qb.DocType("Salary Structure Assignment")
+		employees_with_assignments = (
+			frappe.qb.from_(Assignment)
+			.select(Assignment.employee)
+			.distinct()
+			.where((Assignment.from_date == self.from_date) & (Assignment.docstatus == 1))
+			.run(pluck=True)
+		)
+
+		quick_filter_fields = [
+			"company",
+			"employment_type",
+			"branch",
+			"department",
+			"designation",
+			"grade",
+		]
+		quick_filters = [[d, "=", self.get(d)] for d in quick_filter_fields if self.get(d)]
+
+		filters = (
+			[["status", "=", "Active"], ["employee", "not in", employees_with_assignments]]
+			+ quick_filters
+			+ advanced_filters
+		)
+
+		Employee = frappe.qb.DocType("Employee")
+		Grade = frappe.qb.DocType("Employee Grade")
+		query = (
+			frappe.qb.get_query(
+				Employee, fields=[Employee.employee, Employee.employee_name, Employee.grade], filters=filters
+			)
+			.left_join(Grade)
+			.on(Employee.grade == Grade.name)
+			.select(
+				fn.Coalesce(Grade.default_base_pay, 0).as_("base"),
+				fn.Coalesce(0).as_("variable"),
```

Comment:
You can use `ConstantColumn` instead
https://frappeframework.com/docs/user/en/api/query-builder#constant-column